addons/web/static/src/js/views/list_view.js

```diff
@@ -731,6 +772,14 @@ var ListView = View.extend( /** @lends instance.web.ListView# */ {
     do_add_record: function () {
         this.select_record(null);
     },
+    do_archive_unarchive_selected: function() {
+        var ids = this.groups.get_selection().ids;
+        if (ids.length) {
+            this.do_archive_unarchive(ids);
```

Comment:
pass `records` instead of `ids` so in do_archive_unarchive no need to iterate over ids get record again